sdk/azidentity/managed_identity_client.go

```diff
@@ -191,6 +203,56 @@ func (c *managedIdentityClient) createAppServiceAuthRequest(ctx context.Context,
 	return request, nil
 }
 
+func (c *managedIdentityClient) getAzureArcSecretKey(ctx context.Context, scopes []string) (string, error) {
+	// create the request to retreive the secret key challenge provided by the HIMDS service
+	request, err := azcore.NewRequest(ctx, http.MethodGet, c.endpoint)
+	if err != nil {
+		return "", err
+	}
+	request.Header.Set(azcore.HeaderMetadata, "true")
+	q := request.URL.Query()
+	q.Add("api-version", azureArcAPIVersion)
+	q.Add("resource", strings.Join(scopes, " "))
+	request.URL.RawQuery = q.Encode()
+	// send the initial request to get the short-lived secret key
+	response, err := c.pipeline.Do(request)
+	if err != nil {
+		return "", err
+	}
+	// the endpoint is expected to return a 401 with the WWW-Authenticte header set to the location
+	// of the secret key file. Any other status code indicates an error in the request.
+	if response.StatusCode != 401 {
+		return "", &AuthenticationFailedError{inner: newAADAuthenticationFailedError(response), msg: fmt.Sprintf("Expected a 401 Unauthorized response, received: %d", response.StatusCode)}
+	}
+	if header := response.Header.Get("WWW-Authenticate"); len(header) != 0 {
+		// the WWW-Authenticate header is expected in the following format: Basic realm=/some/file/path.key
```

Comment:
I'd invert the above condition to keep all of this out of the `if` block body.

Reply:
Done